integration/update_test.go

```diff
@@ -673,3 +677,211 @@ func TestUpdateMany(t *testing.T) {
 		})
 	}
 }
+
+func TestCurrentDate(t *testing.T) {
+	t.Parallel()
+
+	// maxTimeDelta is a maximum amount of seconds can differ the value in placeholder from actual value
+	maxTimeDelta := time.Duration(30 * time.Second)
+
+	now := primitive.NewDateTimeFromTime(time.Now().UTC())
+	nowTimestamp := primitive.Timestamp{T: uint32(time.Now().UTC().Unix()), I: uint32(0)}
+
+	for name, tc := range map[string]struct {
+		id     string
+		update bson.D
+		result bson.D
+		paths  []types.Path
+		err    *mongo.WriteError
+		stat   *mongo.UpdateResult
+		alt    string
+	}{
+		"DocumentEmpty": {
+			id:     "double",
+			update: bson.D{{"$currentDate", bson.D{}}},
+			stat: &mongo.UpdateResult{
+				MatchedCount:  1,
+				ModifiedCount: 0,
+				UpsertedCount: 0,
+			},
+			result: bson.D{{"_id", "double"}, {"value", float64(42.13)}},
+		},
+		"Array": {
+			id:     "double",
+			update: bson.D{{"$currentDate", bson.A{}}},
+			err: &mongo.WriteError{
+				Code: 9,
+				Message: "Modifiers operate on fields but we found type array instead. " +
+					"For example: {$mod: {<field>: ...}} not {$currentDate: []}",
+			},
+			alt: "Modifiers operate on fields but we found another type instead",
+		},
+		"WrongInt32": {
+			id:     "double",
+			update: bson.D{{"$currentDate", int32(1)}},
+			err: &mongo.WriteError{
+				Code: 9,
+				Message: "Modifiers operate on fields but we found type int instead. " +
+					"For example: {$mod: {<field>: ...}} not {$currentDate: 1}",
+			},
+			alt: "Modifiers operate on fields but we found another type instead",
+		},
+		"Nil": {
+			id:     "double",
+			update: bson.D{{"$currentDate", nil}},
+			err: &mongo.WriteError{
+				Code: 9,
+				Message: "Modifiers operate on fields but we found type null instead. " +
+					"For example: {$mod: {<field>: ...}} not {$currentDate: null}",
+			},
+			alt: "Modifiers operate on fields but we found another type instead",
+		},
+		"True": {
+			id:     "double",
+			update: bson.D{{"$currentDate", bson.D{{"value", true}}}},
+			stat: &mongo.UpdateResult{
+				MatchedCount:  1,
+				ModifiedCount: 1,
+				UpsertedCount: 0,
+			},
+			paths:  []types.Path{types.NewPathFromString("value")},
+			result: bson.D{{"_id", "double"}, {"value", now}},
+		},
+		"TwoTrue": {
+			id:     "double",
+			update: bson.D{{"$currentDate", bson.D{{"value", true}, {"unexistent", true}}}},
+			stat: &mongo.UpdateResult{
+				MatchedCount:  1,
+				ModifiedCount: 1,
+				UpsertedCount: 0,
+			},
+			paths: []types.Path{
+				types.NewPathFromString("value"),
+				types.NewPathFromString("unexistent"),
+			},
+			result: bson.D{{"_id", "double"}, {"value", now}, {"unexistent", now}},
+		},
+		"False": {
+			id:     "double",
+			update: bson.D{{"$currentDate", bson.D{{"value", false}}}},
```

Comment:
So, both `true` and `false` are valid (I agree).

Does it make sense to add more corner case like `bson.D{{"$currentDate", bson.D{{"value", nil}}}}` and `bson.D{{"$currentDate", bson.D{{"", true}}}}`?

Reply:
bson.D{{"$currentDate", bson.D{{"value", nil}}}}

Reply:
> Does it make sense to add more corner case like `bson.D{{"$currentDate", bson.D{{"value", nil}}}}` 

This test works as expected: no `$type' field gives an error.


> and `bson.D{{"$currentDate", bson.D{{"", true}}}}`?

This test case has been added in #661 issue.